### PR TITLE
Docs: Adding a Markdown Style Guide

### LINKED ISSUE #19592: Docs: Add Markdown style guide
**What would you like to be added**:

A style guide for Markdown content. Here are some things I think would be helpful to start with:

# Markdown style guide

## Styling

- To make text **bold**, use double asterisks, e.g. `**Bold**`.
- To _emphasize_ text, use single underscores, e.g. `_Emphasized_`.
- Punctuation should follow the same styling as the text, e.g. `**TODO:**`, not `**TODO**:`.

### Numbered lists

To avoid inconsistent list numbering, use _lazy list numbering_:

```
1. First
1. Second
1. Third
```

The list above will display as:

```
1. First
2. Second
3. Third
```

**Why is this needed**:

To make documentation and READMEs more consistent and maintainable.

### PR DESCRIPTION
**Pull request by RobinJoe**
What this PR does / why we need it: This request builds a markdown style guide for writers documenting information about Grafana services for users.
This PR serves the need to make documentation and READMEs more consistent and maintainable.

Which issue(s) this PR fixes: This request fixes the issue grafana#19592

Fixes grafana#19592

Special notes for your reviewer:

@marcusolsson and @oddlittlebird, sorry for the delay. I've set up the style guide, and it is ready for review, or additional sections.

**What this PR does / why we need it**: This request builds a markdown style guide for writers documenting information about Grafana services for users.
This PR serves the need to make documentation and READMEs more consistent and maintainable.

**Which issue(s) this PR fixes**: This request fixes the issue grafana#19592

Fixes grafana#19592

**Special notes for your reviewer**:

@marcusolsson and @oddlittlebird, sorry for the delay. I've set up the style guide, and it is ready for review, or additional sections.

### CODE REVIEW

File: contribute/style-guides/documentation-markdown-guide.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+When writing and marking up text with bold or emphasis markdown, include any
+punctuation close to the target text within the markdown:
+
+**Example** **TODO:**, not **TODO**:.
+
+**Example** The following emoji are considered _acceptable:_ Radio buttons, tick marks, and heavy plus signs.
+
+**Example** Emoji became popular in the **mid-2000's**
+
+## Links and References
```
Comment (oddlittlebird) [inline contribute/style-guides/documentation-markdown-guide.md]: Per the documentation style guide, all headings should be in sentence case. Initial cap only.

File: contribute/style-guides/documentation-markdown-guide.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+Create links to other website by wrapping the display text in square brackets, and
+the web URL in curved brackets.
+
+\[text to display](www.website.com)
+
+**Example** For more information on including emoji in Github Flavored Markdown, refer to the [webfx page on emoji](https://www.webfx.com/tools/emoji-cheat-sheet/) for a list of emoji.
```
Comment (oddlittlebird) [inline contribute/style-guides/documentation-markdown-guide.md]:
```suggestion
**Example:** For more information on including emoji in Github Flavored Markdown, refer to the [webfx page on emoji](https://www.webfx.com/tools/emoji-cheat-sheet/) for a list of emoji.
```

File: contribute/style-guides/documentation-markdown-guide.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+## Blockquotes
```
Comment (oddlittlebird) [inline contribute/style-guides/documentation-markdown-guide.md]: Block quotes is two words.

File: contribute/style-guides/documentation-markdown-guide.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+Include Blockquotes inside text using right facing arrows:
```
Comment (oddlittlebird) [inline contribute/style-guides/documentation-markdown-guide.md]:
```suggestion
Include block quotes inside text using right-facing arrows:
```

File: contribute/style-guides/documentation-markdown-guide.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  if (a > 0) {
+    return "positive";
+  } else {
+    return "NOT positive";
+  }
+}
+```
+
+Write the name of the language after the first set of back tics, no spaces,
+to show specific syntax highlighting - for example "\```javascript" produces the following
```
Comment (oddlittlebird) [inline contribute/style-guides/documentation-markdown-guide.md]:
```suggestion
to show specific syntax highlighting. For example: "\```javascript" produces the following
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

**Comment by RobinJoe** [comment]
Looking at the latest commit id (https://github.com/grafana/grafana/pull/22425/commits/224ee8aff4a766ea5f5c7f183212bbc53ed6419d) the content does appear to be a PR aimed to merge into Grafana/Grafana repo. I compared it to other commits tagged with pr/external. I made the requested changes in the latest commit if that helps clarify what stage the work is up to.

**Review by oddlittlebird — APPROVED** [review]

**Comment by oddlittlebird** [comment]
LGTM! Thanks for contributing this resource!

**Pull request merged**